src/setfit/data.py

```diff
@@ -277,6 +277,7 @@ def collate_fn(batch):
 
         # convert to tensors
         features = {k: torch.Tensor(v).int() for k, v in features.items()}
-        labels = torch.Tensor(labels).long()
+        labels = torch.Tensor(labels)
+        labels = labels.long() if isinstance(label, int) else labels.float()
```

Comment:
my suggestion is to use a type of label

The type of label should be 'int' for the single label classification, but 'List' for multilabel classification.

we can write
```
label =  torch.Tensor(labels).long() if isinstance(label, int) else  torch.Tensor(labels).float()
```
but I felt it is too long so that I fix as I push.

Reply:
Or
```python
labels = labels.long() if len(labels[0])== 1 else labels.float()
```

whichever you want!

Reply:
I think the second solution is best! 
```python
labels = labels.long() if len(labels.size()) == 1 else labels.float()
```
That should accurately measure whether we are in a multitarget situation, even if the user accidentally supplies floats instead of integers.

Reply:
Oh, I see.

your suggestion makes sense to me!
I will fix that!